src/main/java/uk/gov/justice/digital/delius/ldap/repository/entity/NDeliusUser.java

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.delius.ldap.repository.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.ldap.odm.annotations.Transient;
+import org.springframework.ldap.odm.annotations.Entry;
+import org.springframework.ldap.odm.annotations.Id;
+
+import javax.naming.Name;
+import java.util.List;
+
+@Data
+@Entry(objectClasses = "NDUser")
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class NDeliusUser {
+    @Id
+    private Name dn;
+    private String cn;
+    private String sn;
+    private String mail;
+    private String givenname;
+    private String orclActiveEndDate;
+    @Transient
```

Comment:
Is this @Transient because roles come from LDAP not the database? So a persist/save never tries to write roles. Does hte database not contain roles? Just a question for my understanding.

Reply:
They do but they are not directly in the same LDAP tree, you have to do a separate ldap search to get them . They are set as aliases to other objects. So we load the NDeliusUser and then fix up the roles using a second call (Spring ldapTemplate would fail if it wasn't transient)